plugins/playstation/src/playstation-header.vala

```diff
@@ -0,0 +1,102 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class Games.PlayStationHeader : Object {
+	private const size_t[] OFFSETS = { 0xD368, 0xD3A8 };
+	private const string[] IDS = { "SLUS", "SCUS", "SLES", "SCES", "SLPS", "SLPM", "SCPS" };
+	private const string MAGIC_VALUE = "BOOT";
+
+  private File cue_file;
+  private File bin_file;
+	private string parent_directory;
+
+	private size_t? header_offset;
+
+	private string _product_number;
+	public string product_number {
+		get {
+			return _product_number;
+		}
+	}
+
+  public PlayStationHeader (File bin_file) {
+    this.bin_file = bin_file;
+		parent_directory = bin_file.get_parent ().get_path () + "/";
+  }
+
+  public void check_validity () throws PlayStationError {
+		foreach (var offset in OFFSETS) {
+			var magic = get_string_at_offset_for_size (offset, MAGIC_VALUE.length);
+			if (magic != MAGIC_VALUE)
+				continue;
+
+			var data = get_string_at_offset (offset);
+
+			foreach (var id in IDS) {
+				if (data.contains (id)) {
+					string raw_id = data.split (id)[1].split (";")[0];
+					raw_id = raw_id.replace ("_", "-");
+					raw_id = raw_id.replace (".", "");
+					_product_number = raw_id;
+					return;
```

Comment:
This `return` isn't the first of its block so there should be newline before it, same for throw, continue, break.